<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>VSV Genomes &amp; Geography - Overview</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css">

</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">VSV Genomes &amp; Geography</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./overview.html" aria-current="page">
 <span class="menu-text">Overview</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./sequences.html">
 <span class="menu-text">Genomes</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://geoepi.github.io/">
 <span class="menu-text">GeoEpi</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://geoepi.github.io/pindex.html">
 <span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/geoepi/vs-epizootics"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Page Contents</h2>
   
  <ul class="collapse">
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link active" data-scroll-target="#load-libraries">Load Libraries</a></li>
  <li><a href="#custom-functions" id="toc-custom-functions" class="nav-link" data-scroll-target="#custom-functions">Custom Functions</a></li>
  <li><a href="#get-boundaries" id="toc-get-boundaries" class="nav-link" data-scroll-target="#get-boundaries">Get Boundaries</a></li>
  <li><a href="#simulate-outbreak" id="toc-simulate-outbreak" class="nav-link" data-scroll-target="#simulate-outbreak">Simulate Outbreak</a></li>
  <li><a href="#spatial-triangulation" id="toc-spatial-triangulation" class="nav-link" data-scroll-target="#spatial-triangulation">Spatial Triangulation</a></li>
  <li><a href="#cluster-distance" id="toc-cluster-distance" class="nav-link" data-scroll-target="#cluster-distance">Cluster Distance</a></li>
  <li><a href="#phylodynamics" id="toc-phylodynamics" class="nav-link" data-scroll-target="#phylodynamics">Phylodynamics</a></li>
  <li><a href="#host-exposure" id="toc-host-exposure" class="nav-link" data-scroll-target="#host-exposure">Host Exposure</a></li>
  <li><a href="#time-index" id="toc-time-index" class="nav-link" data-scroll-target="#time-index">Time Index</a></li>
  <li><a href="#split-predictor-variable" id="toc-split-predictor-variable" class="nav-link" data-scroll-target="#split-predictor-variable">Split Predictor Variable</a></li>
  <li><a href="#projection" id="toc-projection" class="nav-link" data-scroll-target="#projection">Projection</a></li>
  <li><a href="#model-formula" id="toc-model-formula" class="nav-link" data-scroll-target="#model-formula">Model Formula</a></li>
  <li><a href="#run-model" id="toc-run-model" class="nav-link" data-scroll-target="#run-model">Run Model</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Overview</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<style type="text/css">

body, td {
   font-size: 13pt;
}
code.r{
  font-size: 9pt;
}
pre {
  font-size: 11pt
}
</style>
<div class="callout-note callout callout-style-default no-icon callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Supporting Information
</div>
</div>
<div class="callout-body-container callout-body">
<p>This site provides a demonstrative workflow and supporting information for the publication:<br>
<strong>Interrogating Genomes and Geography to Unravel Multiyear Vesicular Stomatitis Epizootics</strong></p>
<p>Humphreys, Shults, Velazquez Salinas, Bertram, Pelzel-McCluskey, Peters, and Rodriguez</p>
<p><strong>Viruses</strong>. 2024; 16(7):1118. <a href="https://doi.org/10.3390/v16071118">https://doi.org/10.3390/v16071118</a></p>
</div>
</div>
<div class="callout-note callout callout-style-default no-icon callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
External Resources
</div>
</div>
<div class="callout-body-container callout-body">
<p>All code and data are available for download:<br>
<a href="https://github.com/geoepi/vs-epizootics">GitHub site: https://github.com/geoepi/vs-epizootics</a></p>
<p>Additional supporting information and documents available on the Open Science Framework:<br>
<a href="https://osf.io/ghzfq/">Project OSF site: https://osf.io/ghzfq/</a></p>
</div>
</div>
<section id="navigation" class="level3">
<h3 class="anchored" data-anchor-id="navigation">Navigation</h3>
<p>The menu at right can be used to navigate through the workflow.</p>
</section>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load Libraries</h2>
<p>Load needed R-packages to execute code. Note that the r-INLA package (<span class="citation" data-cites="Rue2009">Rue, Martino, and Chopin (<a href="#ref-Rue2009" role="doc-biblioref">2009</a>)</span>, <span class="citation" data-cites="Lindgren_2015">Lindgren and Rue (<a href="#ref-Lindgren_2015" role="doc-biblioref">2015</a>)</span>) is not available on CRAN and must be installed from the INLA team website.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyterra)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raptr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rgeos)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pals)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>select <span class="ot">&lt;-</span> dplyr<span class="sc">::</span>select</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(INLA)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("INLA",repos=c(getOption("repos"),INLA="https://inla.r-inla-download.org/R/stable"), dep=TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="custom-functions" class="level2">
<h2 class="anchored" data-anchor-id="custom-functions">Custom Functions</h2>
<p>Custom functions to perform data process and create figure plots. These are available from the linked repository at the top of the script.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1976</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"./R/utilities.R"</span>))</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source_dir</span>(<span class="st">"./R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="get-boundaries" class="level2">
<h2 class="anchored" data-anchor-id="get-boundaries">Get Boundaries</h2>
<p>The <em>assets</em> folder includes raster and other data files to perform the analysis</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>domain <span class="ot">&lt;-</span> <span class="fu">vect</span>(<span class="fu">here</span>(<span class="st">"assets/domain"</span>)) <span class="co"># geographic boundary polygons</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>obs_stack.r <span class="ot">&lt;-</span> <span class="fu">rast</span>( <span class="co"># Rasters depicting Spatial Random Fields estimated from true data</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rast</span>(<span class="fu">here</span>(<span class="st">"assets/rf_2014.tif"</span>)),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rast</span>(<span class="fu">here</span>(<span class="st">"assets/rf_2015.tif"</span>))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>       )</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="simulate-outbreak" class="level2">
<h2 class="anchored" data-anchor-id="simulate-outbreak">Simulate Outbreak</h2>
<p>Virus and disease detection data presented in the manuscript includes the geographic coordinates recording locations for private residences and businesses. This data is confidential and cannot be shared, therefore, this demonstration first simulates outbreak count data from an SEIR-vector model and then randomly assigns geographic locations based on the Spatial Random Fields estimated from the true data. This information is included in the raster files loaded in the previous code chunk.</p>
<section id="seir-vector-dynamics" class="level3">
<h3 class="anchored" data-anchor-id="seir-vector-dynamics">SEIR-Vector Dynamics</h3>
<p>Simulate Vesicular stomatitis disease detections for 2014.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>seir_vector_14 <span class="ot">&lt;-</span> <span class="fu">seirsei_ode</span>(<span class="at">Nh =</span> <span class="dv">4100</span>, <span class="co"># number of livestock hosts</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">vect_mult =</span> <span class="dv">250</span>, <span class="co"># number of vectors per host</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">obs_bias =</span> <span class="fl">0.10</span>, <span class="co"># proportion observed/detected</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">targ_year =</span> <span class="st">"2014"</span> <span class="co"># to approximate dates </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                           )</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seir_vector_14<span class="sc">$</span>adj_inc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 606</code></pre>
</div>
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_seirsei</span>(seir_vector_14)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-seir14" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-seir14-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Simulated VSV outbreak for 2014 with the number of susceptible (Sh), exposed (Eh), infectious (Ih), and removed (Rh) individuals.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Repeat for the 2015 outbreak year.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>seir_vector_15 <span class="ot">&lt;-</span> <span class="fu">seirsei_ode</span>(<span class="at">Nh =</span> <span class="dv">5000</span>, <span class="co"># number of livestock hosts</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">vect_mult =</span> <span class="dv">250</span>, <span class="co"># number of vectors per host</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">obs_bias =</span> <span class="fl">0.10</span>, <span class="co"># proportion observed/detected</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">targ_year =</span> <span class="st">"2015"</span> <span class="co"># to approximate dates </span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                           )</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(seir_vector_15<span class="sc">$</span>adj_inc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 736</code></pre>
</div>
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_seirsei</span>(seir_vector_15)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-seir15" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-seir15-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: Simulated VSV outbreak for 2015 with the number of susceptible (Sh), exposed (Eh), infectious (Ih), and removed (Rh) individuals.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="spatial-randomization" class="level3">
<h3 class="anchored" data-anchor-id="spatial-randomization">Spatial Randomization</h3>
<p><strong>NOTE:</strong> The data produced through simulation appears similar to that presented in the publication, but is different in several regards. Because of these differences, results presented here will not exactly match those presented in the paper.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sample_2014 <span class="ot">&lt;-</span> <span class="fu">spatial_sample</span>(<span class="at">inc_df =</span> seir_vector_14, <span class="at">obs_rast =</span> obs_stack.r[[<span class="dv">1</span>]])</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>sample_2015 <span class="ot">&lt;-</span> <span class="fu">spatial_sample</span>(<span class="at">inc_df =</span> seir_vector_15, <span class="at">obs_rast =</span> obs_stack.r[[<span class="dv">2</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>View the simulated VS cases in relation to</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>join_sims <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sample_2014, sample_2015)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_rast_panels</span>(obs_stack.r, domain, join_sims, <span class="fu">brewer.ylorbr</span>(<span class="dv">4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-simulation" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-simulation-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Simulated disease detections overlying Spatial Random Fields estimated from true data as presented in the publication. Note that although clustering is evident, it is less pronounced as that in true data.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="spatial-triangulation" class="level2">
<h2 class="anchored" data-anchor-id="spatial-triangulation">Spatial Triangulation</h2>
<p>Construct a triangulated mesh to apply Stochastic Partial Differential Equations (see, <span class="citation" data-cites="SPDEbook">Krainski et al. (<a href="#ref-SPDEbook" role="doc-biblioref">2018</a>)</span>). This mesh is at more coarse resolution than that used for the actual analysis in effort to reduce computational demand.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>hull <span class="ot">&lt;-</span> <span class="fu">st_convex_hull</span>(</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">st_buffer</span>(</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">st_union</span>(</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">st_as_sf</span>(domain)),</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="dv">100</span>))</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>dom_bnds <span class="ot">&lt;-</span> <span class="fu">inla.sp2segment</span>(<span class="fu">as</span>(hull, <span class="st">"Spatial"</span>))</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1976</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>mesh.dom <span class="ot">&lt;-</span> <span class="fu">inla.mesh.2d</span>(<span class="at">boundary =</span> dom_bnds, </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">loc =</span> join_sims[,<span class="fu">c</span>(<span class="st">"x"</span>,<span class="st">"y"</span>)],</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cutoff =</span> <span class="dv">20</span>, </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>                        <span class="at">max.edge =</span> <span class="fu">c</span>(<span class="dv">60</span>, <span class="dv">300</span>),</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>                        <span class="at">offset =</span> <span class="fu">c</span>(<span class="dv">100</span>,<span class="dv">250</span>),</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min.angle =</span> <span class="dv">30</span>) </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mesh</span>(mesh.dom)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-mesh" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-mesh-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;4: Trianglulated mesh with vvertices aligned to simulated disease locations. Note that vertex intersections represent nodes.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<section id="node-coordinates" class="level3">
<h3 class="anchored" data-anchor-id="node-coordinates">Node Coordinates</h3>
<p>Extracting the geographic coordinates for node locations. These are used for model integration and represent background characteristics with respect to environmental conditions.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>dd <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(mesh.dom<span class="sc">$</span>loc[,<span class="dv">1</span>], </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                         mesh.dom<span class="sc">$</span>loc[,<span class="dv">2</span>]))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dd) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"x"</span>, <span class="st">"y"</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>dd<span class="sc">$</span>set <span class="ot">&lt;-</span> <span class="st">"node"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<div id="tbl-node_df" class="anchored">
<table class="table table-sm table-striped">
<caption>Table&nbsp;1: Quick look at node data attributes</caption>
<thead>
<tr class="header">
<th style="text-align: right;">x</th>
<th style="text-align: right;">y</th>
<th style="text-align: left;">set</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">-12062.23</td>
<td style="text-align: right;">2800.044</td>
<td style="text-align: left;">node</td>
</tr>
<tr class="even">
<td style="text-align: right;">-12082.84</td>
<td style="text-align: right;">2801.702</td>
<td style="text-align: left;">node</td>
</tr>
<tr class="odd">
<td style="text-align: right;">-12102.67</td>
<td style="text-align: right;">2807.565</td>
<td style="text-align: left;">node</td>
</tr>
<tr class="even">
<td style="text-align: right;">-12179.32</td>
<td style="text-align: right;">2844.257</td>
<td style="text-align: left;">node</td>
</tr>
<tr class="odd">
<td style="text-align: right;">-12219.55</td>
<td style="text-align: right;">2868.210</td>
<td style="text-align: left;">node</td>
</tr>
<tr class="even">
<td style="text-align: right;">-12236.23</td>
<td style="text-align: right;">2880.891</td>
<td style="text-align: left;">node</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</section>
<section id="copy-to-year" class="level3">
<h3 class="anchored" data-anchor-id="copy-to-year">Copy to Year</h3>
<p>Create a separate copy of the nodes for each year of interest, join the nodes with simulated outbreaks, and assign a random calendar date to each. The nodes are joined to the observation data where a fictitious predictor variable is created (<em>pred_var</em>) and a column obs is added to code a 1 for disease observations and a 0 for mesh nodes. Note that the <em>pred_var</em> are simulated using different mean values so that years show a different average. This process is performed for each year separately.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>seq_2014_dates <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">as.Date</span>(<span class="st">"2014-01-01"</span>), <span class="fu">as.Date</span>(<span class="st">"2014-12-31"</span>), <span class="at">by =</span> <span class="dv">1</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>sample_2014 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sample_2014, <span class="fu">mutate</span>(dd, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">year =</span> <span class="dv">2014</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">date =</span> <span class="fu">sample</span>(seq_2014_dates, <span class="fu">n</span>(), <span class="at">replace =</span> <span class="cn">TRUE</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pred_var =</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">obs =</span> <span class="fu">ifelse</span>(set <span class="sc">==</span> <span class="st">"node"</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>seq_2015_dates <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">as.Date</span>(<span class="st">"2015-01-01"</span>), <span class="fu">as.Date</span>(<span class="st">"2015-12-31"</span>), <span class="at">by =</span> <span class="dv">1</span>)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>sample_2015 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sample_2015, <span class="fu">mutate</span>(dd, </span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">year =</span> <span class="dv">2015</span>,</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">date =</span> <span class="fu">sample</span>(seq_2015_dates, <span class="fu">n</span>(), <span class="at">replace =</span> <span class="cn">TRUE</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">pred_var =</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">mean =</span> <span class="fl">0.5</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">obs =</span> <span class="fu">ifelse</span>(set <span class="sc">==</span> <span class="st">"node"</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample_2015)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<div id="tbl-samp_df" class="anchored">
<table class="table table-sm table-striped">
<caption>Table&nbsp;2: Quick look at resulting data attributes</caption>
<thead>
<tr class="header">
<th style="text-align: right;">x</th>
<th style="text-align: right;">y</th>
<th style="text-align: right;">year</th>
<th style="text-align: left;">set</th>
<th style="text-align: left;">date</th>
<th style="text-align: right;">pred_var</th>
<th style="text-align: right;">obs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">-11830.07</td>
<td style="text-align: right;">4146.953</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-03-27</td>
<td style="text-align: right;">-0.5322010</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">-11977.28</td>
<td style="text-align: right;">4739.053</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-03-28</td>
<td style="text-align: right;">0.6199045</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">-11830.07</td>
<td style="text-align: right;">5292.953</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-03-29</td>
<td style="text-align: right;">3.8629048</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">-11745.95</td>
<td style="text-align: right;">4605.353</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-03-30</td>
<td style="text-align: right;">2.0648198</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: right;">-11991.30</td>
<td style="text-align: right;">4643.553</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-03-31</td>
<td style="text-align: right;">0.6998745</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">-11472.56</td>
<td style="text-align: right;">5464.853</td>
<td style="text-align: right;">2015</td>
<td style="text-align: left;">sim</td>
<td style="text-align: left;">2015-04-01</td>
<td style="text-align: right;">2.0260718</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</section>
</section>
<section id="cluster-distance" class="level2">
<h2 class="anchored" data-anchor-id="cluster-distance">Cluster Distance</h2>
<p>A constructed covariate, or <em>cluster</em> variable is made to account for spatial correlation below the resolution of the mesh. This is done by calculating the distance to nearest locations with a simulated disease detection. (see, <span class="citation" data-cites="Illian_2012">Illian, Sørbye, and Rue (<a href="#ref-Illian_2012" role="doc-biblioref">2012</a>)</span> for discussion of this approach)</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 2014</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>split_factor <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(sample_2014<span class="sc">$</span>set)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>NN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  spatstat.geom<span class="sc">::</span><span class="fu">nndist</span>(sample_2014[,<span class="fu">c</span>(<span class="st">"x"</span>,<span class="st">"y"</span>)], <span class="at">by =</span> split_factor, <span class="at">k =</span> <span class="dv">1</span>) </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>)[<span class="st">"sim"</span>]</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>sample_2014<span class="sc">$</span>NN <span class="ot">&lt;-</span> <span class="fu">round</span>(NN<span class="sc">$</span>sim, <span class="dv">0</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 2015</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>split_factor <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(sample_2015<span class="sc">$</span>set)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>NN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  spatstat.geom<span class="sc">::</span><span class="fu">nndist</span>(sample_2015[,<span class="fu">c</span>(<span class="st">"x"</span>,<span class="st">"y"</span>)], <span class="at">by =</span> split_factor, <span class="at">k =</span> <span class="dv">1</span>)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>)[<span class="st">"sim"</span>]</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>sample_2015<span class="sc">$</span>NN <span class="ot">&lt;-</span> <span class="fu">round</span>(NN<span class="sc">$</span>sim, <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="combine-data" class="level3">
<h3 class="anchored" data-anchor-id="combine-data">Combine Data</h3>
<p>Joining data from both 2014 and 2015 to a common data set.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">=</span> <span class="fu">rbind</span>(sample_2014, sample_2015)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="phylodynamics" class="level2">
<h2 class="anchored" data-anchor-id="phylodynamics">Phylodynamics</h2>
<p>A maximum clade credibility tree is loaded. This tree includes actual disease genomic data but does not have location information. After reading the phylogeny, phylodynamic analysis is performed using the <strong>phylodyn</strong> package (<span class="citation" data-cites="Karcher_2017">Karcher et al. (<a href="#ref-Karcher_2017" role="doc-biblioref">2017</a>)</span>). The resulting effective population size estimate (<strong>Ne</strong>) is matched to the data frame by date.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>vsv_tree <span class="ot">=</span> <span class="fu">read.tree</span>(<span class="fu">here</span>(<span class="st">"assets/mcc.tre"</span>))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>phylodynamics <span class="ot">&lt;-</span> <span class="fu">phylodynamic_process</span>(vsv_tree)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(phylodynamics<span class="sc">$</span>dyn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">date</th>
<th style="text-align: right;">Ne</th>
<th style="text-align: right;">Ne.low</th>
<th style="text-align: right;">Ne.high</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">2016-01-09</td>
<td style="text-align: right;">0.6075437</td>
<td style="text-align: right;">0.1530336</td>
<td style="text-align: right;">2.305747</td>
</tr>
<tr class="even">
<td style="text-align: left;">2016-01-07</td>
<td style="text-align: right;">0.5598360</td>
<td style="text-align: right;">0.1489246</td>
<td style="text-align: right;">1.982533</td>
</tr>
<tr class="odd">
<td style="text-align: left;">2016-01-04</td>
<td style="text-align: right;">0.5197036</td>
<td style="text-align: right;">0.1439427</td>
<td style="text-align: right;">1.750127</td>
</tr>
<tr class="even">
<td style="text-align: left;">2016-01-02</td>
<td style="text-align: right;">0.4856708</td>
<td style="text-align: right;">0.1385381</td>
<td style="text-align: right;">1.575660</td>
</tr>
<tr class="odd">
<td style="text-align: left;">2015-12-31</td>
<td style="text-align: right;">0.4565942</td>
<td style="text-align: right;">0.1330237</td>
<td style="text-align: right;">1.439976</td>
</tr>
<tr class="even">
<td style="text-align: left;">2015-12-29</td>
<td style="text-align: right;">0.4315794</td>
<td style="text-align: right;">0.1276110</td>
<td style="text-align: right;">1.331279</td>
</tr>
</tbody>
</table>
</div>
</div>
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">&lt;-</span> <span class="fu">append_nearest_date</span>(combined_data, phylodynamics<span class="sc">$</span>dyn, <span class="st">"Ne"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(phylodynamics<span class="sc">$</span>tree)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-tree" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-tree-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;5: VSV Phylogeny</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(phylodynamics<span class="sc">$</span>Ne)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-phylodyn" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-phylodyn-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;6: VSV Effective Population size (Ne).</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="host-exposure" class="level2">
<h2 class="anchored" data-anchor-id="host-exposure">Host Exposure</h2>
<p>Because the goal is to estimate disease intensity for a geographic area, not just individual point locations, it is necessary to identify the exposure rate of livestock (disease hosts) to VSV. That is, estimates for host population must be assigned to point locations. Host population, represented here by the <strong>host_pop</strong> raster file, must be added to the data frame. However, coordinates need to represent a geographic area, not just a point, thus a simple point extraction from the raster would be naive. Instead, <em>natural neighborhoods</em> are defined based on node locations, then total host population is summed across this areal extent before adding the value to the node point associated with that neighborhood. This is process is carried out using the <em>mesh_tessellation()</em> function below.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>host_pop.r <span class="ot">&lt;-</span> <span class="fu">rast</span>(<span class="fu">here</span>(<span class="st">"assets/host_pop.tif"</span>))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>tessel_mesh <span class="ot">&lt;-</span> <span class="fu">vect</span>(<span class="fu">mesh_tessellation</span>(mesh.dom))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">crs</span>(tessel_mesh) <span class="ot">&lt;-</span> <span class="fu">crs</span>(domain)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="co"># scale the population</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>tessel_mesh<span class="sc">$</span>host_pop <span class="ot">&lt;-</span> <span class="fu">extract</span>(host_pop.r, tessel_mesh, <span class="at">fun=</span><span class="st">"sum"</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>)[,<span class="st">"host_pop"</span>]<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">3</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>host_pop <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">extract</span>(tessel_mesh, combined_data[,<span class="fu">c</span>(<span class="st">"x"</span>,<span class="st">"y"</span>)])) <span class="sc">%&gt;%</span> </span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(id.y, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) </span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>combined_data<span class="sc">$</span>host_pop <span class="ot">&lt;-</span> host_pop<span class="sc">$</span>host_pop</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_exposure_seq</span>(<span class="at">mesh =</span> mesh.dom, <span class="at">dd_nodes =</span> dd, <span class="at">rast_data =</span> host_pop.r, </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rast_scale =</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">3</span>, <span class="at">xmin =</span> <span class="sc">-</span><span class="dv">11500</span>, <span class="at">xmax =</span> <span class="sc">-</span><span class="dv">10500</span>, <span class="at">ymin =</span> <span class="dv">3500</span>, <span class="at">ymax =</span> <span class="dv">4500</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-exposure" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-exposure-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;7: Plots depict the exposure estimation process with color indicating host abundance(darker tones indicate more hosts). Mesh overlying a continuous host abudnance raster grid (top left), removing mesh edges from plot and calculating natural neighborhoods around nodes using tesselation (top right), removing nodes to show total host abundance summed to neighborhood areas (bottom left), and showing assignment of neighborhood sums back to nodes (bottom right).</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="time-index" class="level2">
<h2 class="anchored" data-anchor-id="time-index">Time Index</h2>
<p>Recoding year and date variables as integer based indices.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>combined_data<span class="sc">$</span>year_step <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.factor</span>(combined_data<span class="sc">$</span>year))</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>combined_data<span class="sc">$</span>day_step <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">as.factor</span>(combined_data<span class="sc">$</span>date))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="split-predictor-variable" class="level2">
<h2 class="anchored" data-anchor-id="split-predictor-variable">Split Predictor Variable</h2>
<p>Ensuring a difference is evident between years within the fictitious prediction variables, which are intended to represent environmental conditions.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">&lt;-</span> combined_data <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pred_var_14 =</span> <span class="fu">if_else</span>(year <span class="sc">==</span> <span class="dv">2014</span>, pred_var, <span class="cn">NA</span>),</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">pred_var_15 =</span> <span class="fu">if_else</span>(year <span class="sc">==</span> <span class="dv">2015</span>, pred_var, <span class="cn">NA</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="projection" class="level2">
<h2 class="anchored" data-anchor-id="projection">Projection</h2>
<section id="random-field" class="level3">
<h3 class="anchored" data-anchor-id="random-field">Random Field</h3>
<p>Coordinate locations in the data frame are matched or <em>projected</em> to the spatial mesh and a corresponding index is created to identify these locations during spatial field estimation.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>k <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(combined_data<span class="sc">$</span>year_step)) <span class="co"># number of years to model, each with a separate spatial field.</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>locs <span class="ot">=</span> <span class="fu">cbind</span>(combined_data<span class="sc">$</span>x, combined_data<span class="sc">$</span>y)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Match locations in data frame to locations in mesh</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>A.pf <span class="ot">=</span> <span class="fu">inla.spde.make.A</span>(mesh.dom, </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">alpha =</span> <span class="dv">2</span>,</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">loc=</span>locs,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> combined_data<span class="sc">$</span>year_step)</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>spde0 <span class="ot">=</span> <span class="fu">inla.spde2.pcmatern</span>(mesh.dom, <span class="at">alpha =</span> <span class="dv">2</span>,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">prior.range=</span><span class="fu">c</span>(<span class="dv">250</span>, <span class="fl">0.01</span>), <span class="co"># a 0.01 probability that auto correlation falls to zero at approximately 250km</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">prior.sigma=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.01</span>),</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">constr =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="co"># index spatial field to have k replicates (2 replicates, one for each year)</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>Field.pf <span class="ot">=</span> <span class="fu">inla.spde.make.index</span>(<span class="st">"Field.pf"</span>, </span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>                               spde0<span class="sc">$</span>n.spde,</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">n.group=</span>k)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="organize-data" class="level3">
<h3 class="anchored" data-anchor-id="organize-data">Organize Data</h3>
<p>First tier of the spatial model is organized as a <em>list()</em> object. This includes an intercept (<span class="math inline">\beta_{\Lambda}</span>) and index for the spatial field (<span class="math inline">\text{W}_{st}\}</span>).</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>pf.lst <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">c</span>(Field.pf, <span class="co"># index for spatial field</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">intercept1 =</span> <span class="dv">1</span>)), <span class="co"># intercept, </span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">year_step1 =</span> combined_data[,<span class="st">"year_step"</span>])) <span class="co"># time index for year (t)</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>pf.stk <span class="ot">=</span> <span class="fu">inla.stack</span>(<span class="at">data =</span> <span class="fu">list</span>(<span class="at">Y =</span> <span class="fu">cbind</span>(combined_data<span class="sc">$</span>obs, <span class="cn">NA</span>)), <span class="co"># creating a two bivariate matrix, </span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">A =</span> <span class="fu">list</span>(A.pf, <span class="dv">1</span>), <span class="co"># projection matrix</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">effects =</span> pf.lst, <span class="co"># data frame </span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">tag =</span> <span class="st">"pf.0"</span>) <span class="co"># label to pull data later</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="estimate-stack" class="level3">
<h3 class="anchored" data-anchor-id="estimate-stack">Estimate Stack</h3>
<p>As in prior chunk, data for the second tier of the model is organized as a <em>list()</em>. This includes an index (<em>Field.st</em>) to copy the spatial field from tier 1 (<span class="math inline">\text{W}_{st}\}</span>) to tier 2, an intercept (<span class="math inline">\beta_0</span>), the fictitious prediction variables (, <em>pred_var</em>, <span class="math inline">\beta_bX_{bst}</span>), cluster variable (<span class="math inline">\gamma_{clust}</span>), effective population size (<span class="math inline">\gamma_{ne}</span>), and needed time indices (t).</p>
<p><strong>NOTE:</strong> Spatial fields are estimated for each year (two time steps), where as effective population size is daily.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>A.est <span class="ot">=</span> <span class="fu">inla.spde.make.A</span>(mesh.dom, <span class="co"># projection as previously done</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">alpha =</span> <span class="dv">2</span>,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">loc=</span>locs,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group =</span> combined_data<span class="sc">$</span>year_step)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>Field.est <span class="ot">=</span> <span class="fu">inla.spde.make.index</span>(<span class="st">"Field.est"</span>, <span class="co"># index spatial filed to copy W_st from tier 1</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>                                 spde0<span class="sc">$</span>n.spde,</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">n.group=</span>k)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>est.lst <span class="ot">=</span> <span class="fu">list</span>(<span class="fu">c</span>(Field.est, <span class="co"># spatial field</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">list</span>(<span class="at">intercept2 =</span> <span class="dv">1</span>)), <span class="co"># level specific intercept</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">list</span>(<span class="at">pred_var_14 =</span> combined_data[,<span class="st">"pred_var_14"</span>], <span class="co"># fictitious fixed/linear effect</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pred_var_15 =</span> combined_data[,<span class="st">"pred_var_15"</span>],</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">NN =</span> combined_data[,<span class="st">"NN"</span>], <span class="co"># cluster distances</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Ne =</span> combined_data[,<span class="st">"Ne"</span>], <span class="co"># effective population size</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">year_step1 =</span> combined_data[,<span class="st">"year_step"</span>], <span class="co"># time indices</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>                      <span class="at">day_step =</span> combined_data[,<span class="st">"day_step"</span>]))</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>est.stk <span class="ot">=</span> <span class="fu">inla.stack</span>(<span class="at">data =</span> <span class="fu">list</span>(<span class="at">Y =</span> <span class="fu">cbind</span>(<span class="cn">NA</span>, combined_data<span class="sc">$</span>obs), <span class="co"># creating a two bivariate matrix</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">e =</span> combined_data<span class="sc">$</span>host_pop), <span class="co"># estimated host exposure</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">A =</span> <span class="fu">list</span>(A.est, <span class="dv">1</span>), </span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">effects =</span> est.lst,   </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>                               <span class="at">tag =</span> <span class="st">"est.0"</span>)</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>joint.stk <span class="ot">=</span> <span class="fu">inla.stack</span>(pf.stk, est.stk) <span class="co"># join Tier 1 and Tier 2 to a common list object</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="model-formula" class="level2">
<h2 class="anchored" data-anchor-id="model-formula">Model Formula</h2>
<p>The model is described fully in the manuscript but is provided below as a reference for comparison to code.</p>
<p><span class="math display">\begin{align}
   \text{log}(\Lambda_{st}) &amp;= W_{st} \nonumber \\ \nonumber
    \quad \quad \Lambda_{st} &amp;= \text{exp}\{\beta_{\Lambda} + \text{W}_{st}\}\\ \nonumber
    \text{W}_{st} &amp;\overset{\textit{iid}}{\sim} \textit{N}(0, \text{Q}(\textit{r},\sigma)) \\ \nonumber
    \textit{r} &amp;\sim \textit{Pr}(250, 0.01) \\ \nonumber
    \sigma &amp;\sim \textit{Pr}(1, 0.01) \\ \nonumber
   \text{Y}_{st}|\lambda_{st}  &amp;\sim \text{Poisson}(\mu_{st})  \nonumber \\ \nonumber
   \mu_{st} &amp;= \textit{E}_{\textit{st}}\lambda_{st} \\ \nonumber
   log(\mu_{\textit{st}}) &amp;= log(\textit{E}_{\textit{st}}) + log(\lambda_{\textit{st}}) \\ \nonumber
   log(\lambda_{\textit{st}}) &amp;=  \beta_0 + \sum_{b=1}^{B} \beta_bX_{bst} + \gamma_{clust} + \gamma_{river} + \gamma_{ne} + \alpha \text{W}_{st}  \\  \nonumber
   \beta_b &amp;= (\beta_{eddi}, \beta_{wet}, \beta_{temp}, \beta_{seas}, \beta_{carb}, \beta_{shrb}, \beta_{elev}, \beta_{ndvi}) \\ \nonumber
   \beta_b &amp;\sim \textit{N}(1, 0.001) \\ \nonumber
   \gamma_{clust}, \gamma_{river} &amp;\sim \textit{Pr}(1, 0.01) \\ \nonumber
   \gamma_{ne} &amp;\sim \textit{Pr}(3, 0.01) \\ \nonumber
\end{align}</span></p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>pcprior1 <span class="ot">=</span> <span class="fu">list</span>(<span class="at">prec =</span> <span class="fu">list</span>(<span class="at">prior=</span><span class="st">"pc.prec"</span>, <span class="at">param =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.01</span>))) </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>hc1 <span class="ot">=</span> <span class="fu">list</span>(<span class="at">theta =</span> <span class="fu">list</span>(<span class="at">prior =</span> <span class="st">'normal'</span>, <span class="at">param =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>)))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>ctr.g <span class="ot">=</span> <span class="fu">list</span>(<span class="at">model =</span> <span class="st">'iid'</span>, <span class="at">hyper =</span> hc1)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>Formula<span class="fl">.1</span> <span class="ot">=</span> Y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> intercept1 <span class="sc">+</span> <span class="co"># intercept tier 1</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                     intercept2 <span class="sc">+</span> <span class="co"># intercept tier 2</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">f</span>(Field.pf, <span class="co"># spatial index for tier 1 spatial field</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">model=</span>spde0,</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> Field.pf.group, <span class="co"># group by year, k=2</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>                            <span class="at">control.group=</span>ctr.g) <span class="sc">+</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">f</span>(Field.est, <span class="co"># create copy of tier 1 spatial field</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">copy =</span> <span class="st">"Field.pf"</span>, </span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> Field.est.group, </span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fixed =</span> <span class="cn">FALSE</span>,</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>                            <span class="at">hyper =</span> hc1) <span class="sc">+</span> <span class="co"># IID between years</span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">f</span>(NN,     <span class="co"># Cluster distance</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">model=</span><span class="st">"rw1"</span>, <span class="co"># order-1 random walk</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">constr=</span><span class="cn">TRUE</span>, <span class="co"># center values</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>                            <span class="at">replicate =</span> year_step1, <span class="co"># estimate independently (IID) for each year</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">scale.model=</span><span class="cn">TRUE</span>, <span class="co"># internal scaling</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>                            <span class="at">hyper=</span>pcprior1) <span class="sc">+</span> <span class="co">#Penalizing complexity priors</span></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">f</span>(day_step, Ne,   <span class="co"># Similar to above, but for effective population size</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">model=</span><span class="st">"rw1"</span>, </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">constr=</span><span class="cn">TRUE</span>,</span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">scale.model=</span><span class="cn">TRUE</span>,</span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>                            <span class="at">hyper=</span>pcprior1) <span class="sc">+</span></span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a>                  pred_var_14 <span class="sc">+</span> pred_var_15 <span class="co"># linear effects </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-model" class="level2">
<h2 class="anchored" data-anchor-id="run-model">Run Model</h2>
<p>This light, demonstrative model took approximately 1hr to run. The saved model is loaded below to examine outputs and results.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>theta1 <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">3.091600</span>, <span class="fl">6.136307</span>, <span class="sc">-</span><span class="fl">2.297550</span>, <span class="sc">-</span><span class="fl">1.228109</span>, <span class="sc">-</span><span class="fl">2.166505</span>, <span class="sc">-</span><span class="fl">1.035271</span>) <span class="co"># mean hyperparameters from initial run (speeds up runs)  </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>Model.demo <span class="ot">&lt;-</span> <span class="fu">inla</span>(Formula<span class="fl">.1</span>, <span class="co"># modle formula</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">num.threads =</span> <span class="dv">12</span>, <span class="co"># threads</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data =</span> <span class="fu">inla.stack.data</span>(joint.stk),  <span class="co"># data organized above as list</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">family =</span> <span class="fu">c</span>(<span class="st">"gaussian"</span>, <span class="st">"poisson"</span>), <span class="co"># tier 1 and tier 2 distribution families</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">E =</span> <span class="fu">inla.stack.data</span>(joint.stk)<span class="sc">$</span>e, <span class="co"># exposure</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control.fixed =</span> <span class="fu">list</span>(<span class="at">prec =</span> <span class="dv">1</span>, <span class="at">prec.intercept=</span><span class="dv">1</span>), <span class="co"># proper intercept</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control.predictor =</span> <span class="fu">list</span>(</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">A =</span> <span class="fu">inla.stack.A</span>(joint.stk), </span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">compute =</span> <span class="cn">TRUE</span>, <span class="co"># calculate fitted values</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">link =</span> <span class="dv">1</span>), <span class="co"># default link functions</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control.mode =</span> <span class="fu">list</span>(<span class="at">restart =</span> <span class="cn">TRUE</span>, <span class="at">theta =</span> theta1), <span class="co"># hyperparameters to speed up run</span></span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control.inla =</span> <span class="fu">list</span>(<span class="at">strategy=</span><span class="st">"adaptive"</span>, <span class="co"># fast efficient stratigy</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">int.strategy =</span> <span class="st">"eb"</span>),</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">control.compute=</span><span class="fu">list</span>(<span class="at">dic =</span> <span class="cn">FALSE</span>, <span class="at">cpo =</span> <span class="cn">FALSE</span>, <span class="at">waic =</span> <span class="cn">FALSE</span>)) </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(<span class="at">list=</span><span class="fu">c</span>(<span class="st">"Model.demo"</span>), <span class="at">file=</span><span class="fu">here</span>(<span class="st">"assets/demo.RData"</span>), <span class="at">version =</span> <span class="dv">2</span>) <span class="co"># save model</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Load previously run model</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"assets/demo.RData"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="results" class="level2">
<h2 class="anchored" data-anchor-id="results">Results</h2>
<section id="create-grid" class="level3">
<h3 class="anchored" data-anchor-id="create-grid">Create Grid</h3>
<p>Creatin a point grid and raster to visulaize results.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>blank.r <span class="ot">&lt;-</span> obs_stack.r[[<span class="dv">1</span>]]</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>blank.r[<span class="sc">!</span><span class="fu">is.na</span>(blank.r)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>grid_pnts <span class="ot">&lt;-</span> <span class="fu">as.points</span>(blank.r)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(grid_pnts) <span class="ot">&lt;-</span> <span class="st">"cell_value"</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>grid_coords <span class="ot">&lt;-</span> grid_pnts <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>Ap <span class="ot">=</span> <span class="fu">inla.spde.make.A</span>(mesh.dom, </span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">loc =</span> <span class="fu">cbind</span>(grid_coords[,<span class="st">"x"</span>], </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>                                  grid_coords[,<span class="st">"y"</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="cluster-distance-1" class="level3">
<h3 class="anchored" data-anchor-id="cluster-distance-1">Cluster Distance</h3>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_cluster</span>(Model.demo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-clust" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-clust-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;8: Spatial clustering of simulated disease detections</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-coefficients" class="level3">
<h3 class="anchored" data-anchor-id="linear-coefficients">Linear Coefficients</h3>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_fixed_marginals</span>(Model.demo, <span class="fu">c</span>(<span class="st">"pred_var_14"</span>, <span class="st">"pred_var_15"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-fixed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-fixed-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;9: Fictitious linear effects constructed with different mean averages.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="phylodynamic-effect" class="level3">
<h3 class="anchored" data-anchor-id="phylodynamic-effect">Phylodynamic Effect</h3>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_phylodynamic</span>(Model.demo, <span class="at">parameter =</span> <span class="st">"day_step"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-phylo_out" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-phylo_out-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;10: Effective population size effect.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="spatial-random-fields" class="level3">
<h3 class="anchored" data-anchor-id="spatial-random-fields">Spatial Random Fields</h3>
<p>Extract estimates from model and rasterize them for visualization.</p>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>mrf_pf <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Model.demo<span class="sc">$</span>summary.random<span class="sc">$</span>Field.pf<span class="sc">$</span>mean, </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>                Field.pf<span class="sc">$</span>Field.pf.group)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>mrf_pf_v <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>mrf_pf_v  <span class="ot">&lt;-</span> <span class="fu">split</span>(mrf_pf[,<span class="dv">1</span>], mrf_pf[,<span class="dv">2</span>])</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>grid_pnts<span class="sc">$</span>pf_2014 <span class="ot">&lt;-</span> <span class="fu">drop</span>(Ap <span class="sc">%*%</span> mrf_pf_v[[<span class="dv">1</span>]]) </span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>grid_pnts<span class="sc">$</span>pf_2015 <span class="ot">&lt;-</span> <span class="fu">drop</span>(Ap <span class="sc">%*%</span> mrf_pf_v[[<span class="dv">2</span>]]) </span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>pf_14.r <span class="ot">&lt;-</span> <span class="fu">rasterize</span>(grid_pnts, </span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>                    blank.r, </span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"pf_2014"</span>,</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background =</span> <span class="cn">NA</span>)</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>pf_15.r <span class="ot">&lt;-</span> <span class="fu">rasterize</span>(grid_pnts, </span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>                    blank.r, </span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"pf_2015"</span>,</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">background =</span> <span class="cn">NA</span>)</span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>pf_stack.r <span class="ot">&lt;-</span><span class="fu">list</span>(pf_14.r, pf_15.r)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="">
<summary>Hide code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_rast_panels</span>(pf_stack.r, domain, join_sims, <span class="fu">rev</span>(<span class="fu">cubehelix</span>(<span class="dv">50</span>)), <span class="at">plot_pnts =</span> <span class="cn">FALSE</span>, <span class="at">center=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-rfs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="overview_files/figure-html/fig-rfs-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;11: Year specific Spatial Random Fields estimated by demonstarted model. Similiar to those used to used for simulation.</figcaption><p></p>
</figure>
</div>
</div>
</div>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-Illian_2012" class="csl-entry" role="doc-biblioentry">
Illian, Janine B., Sigrunn H. Sørbye, and Håvard Rue. 2012. <span>“<span class="nocase">A toolbox for fitting complex spatial point process models using integrated nested Laplace approximation (INLA)</span>.”</span> <em>The Annals of Applied Statistics</em> 6 (4): 1499–1530. <a href="https://doi.org/10.1214/11-AOAS530">https://doi.org/10.1214/11-AOAS530</a>.
</div>
<div id="ref-Karcher_2017" class="csl-entry" role="doc-biblioentry">
Karcher, Michael D., Julia A. Palacios, Shiwei Lan, and Vladimir N. Minin. 2017. <span>“<span class="nocase">phylodyn: an R package for phylodynamic simulation and inference</span>.”</span> <em>Molecular Ecology</em>, 1–9. <a href="https://doi.org/10.1111/1755-0998.12630">https://doi.org/10.1111/1755-0998.12630</a>.
</div>
<div id="ref-SPDEbook" class="csl-entry" role="doc-biblioentry">
Krainski, Elias, Virgilio Gómez Rubio, Haakon Bakka, A Lenzi, Daniela Castro-Camilo, Daniel Simpson, Finn Lindgren, and Håvard Rue. 2018. <em>Advanced Spatial Modeling with Stochastic Partial Differential Equations Using r and INLA</em>. Chapman; Hall/CRC. <a href="https://doi.org/10.1201/9780429031892">https://doi.org/10.1201/9780429031892</a>.
</div>
<div id="ref-Lindgren_2015" class="csl-entry" role="doc-biblioentry">
Lindgren, Finn, and Håvard Rue. 2015. <span>“Bayesian Spatial Modelling with r-INLA.”</span> <em>Journal of Statistical Software</em> 63 (19): 1–25. <a href="https://doi.org/10.18637/jss.v063.i19">https://doi.org/10.18637/jss.v063.i19</a>.
</div>
<div id="ref-Rue2009" class="csl-entry" role="doc-biblioentry">
Rue, Håvard, Sara Martino, and Nicolas Chopin. 2009. <span>“<span class="nocase">Approximate Bayesian inference for latent Gaussian models by using integrated nested Laplace approximations</span>.”</span> <em>Journal of the Royal Statistical Society. Series B: Statistical Methodology</em> 71 (2): 319–92. <a href="https://doi.org/10.1111/j.1467-9868.2008.00700.x">https://doi.org/10.1111/j.1467-9868.2008.00700.x</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">Built with <a href="https://quarto.org/">Quarto</a></div>
  </div>
</footer>



<script src="site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>